src/main/java/frc/robot/Subsystems/ClimberPlacer.java

```diff
@@ -5,46 +5,75 @@
 import edu.wpi.first.wpilibj.DigitalInput;
 import frc.robot.RobotMap;
 
+import java.util.function.Supplier;
+
 /**
  * Controls the position of the climber.
  */
 public class ClimberPlacer extends MotoredGenericSubsystem {
 
+    private final Supplier<Double> STALL_CURRENT = rootNamespace.addConstantDouble("stall current", 0);
+
     private static final double MIN_SPEED = -0.6;
     private static final double MAX_SPEED = 0.6;
 
-    private static ClimberPlacer instance;
+    private static ClimberPlacer rightInstance, leftInstance;
+
+    private final Supplier<Double> MOVEMENT_SPEED = rootNamespace.addConstantDouble("movement speed", 0);
 
     private final DigitalInput frontLimit;
     private final DigitalInput backLimit;
-    private static WPI_TalonSRX leftPlacer;
-    private static WPI_TalonSRX rightPlacer;
-
-    public static ClimberPlacer getInstance() {
-        if (instance == null) {
-            leftPlacer = new WPI_TalonSRX(RobotMap.CAN.PLACER_TALON_LEFT);
-            rightPlacer = new WPI_TalonSRX(RobotMap.CAN.PLACER_TALON_RIGHT);
-            instance = new ClimberPlacer(leftPlacer, rightPlacer);
+    private final WPI_TalonSRX placer;
+
+    private boolean stalling = false;
+    private boolean passedFirstStall = false;
+
+    public static ClimberPlacer getRightInstance() {
+        if (rightInstance == null) {
+            rightInstance = new ClimberPlacer(new WPI_TalonSRX(RobotMap.CAN.PLACER_TALON_RIGHT));
         }
-        return instance;
+        return rightInstance;
     }
 
-    private ClimberPlacer(WPI_TalonSRX leftPlacer, WPI_TalonSRX rightPlacer) {
-        super(MIN_SPEED, MAX_SPEED, "climber placer", leftPlacer, rightPlacer);
+    public static ClimberPlacer getLeftInstance() {
+        if (leftInstance == null) {
+            leftInstance = new ClimberPlacer(new WPI_TalonSRX(RobotMap.CAN.PLACER_TALON_LEFT));
+        }
+        return leftInstance;
+    }
+
+    private ClimberPlacer(WPI_TalonSRX placer) {
+        super(MIN_SPEED, MAX_SPEED, "right climber placer", placer);
+        this.placer = placer;
         this.frontLimit = new DigitalInput(RobotMap.DIO.PLACER_LIMIT_FRONT);
         this.backLimit = new DigitalInput(RobotMap.DIO.PLACER_LIMIT_BACK);
     }
 
+    @Override
+    protected void apply(double speed) {
+        if (!passedFirstStall && placer.getStatorCurrent() > STALL_CURRENT.get())
```

Comment:
Move `placer.getStatorCurrent() > STALL_CURRENT.get()` to seperated function for readability